### PR TITLE
feat: KAN-114-auto-tf-resource

### PR DESCRIPTION
**Pull request by canle144**
BREAKING CHANGE:
- Add new workflow to init TF state resources
- Update workflow to work with new tf module repo
- https://github.com/ib-agent/can5-service/actions
- KAN-114

### CODE REVIEW

**Comment by canle144** [comment]
PR relates to https://github.com/ib-agent/cookiecutter-python-service/pull/6

**Review by brook-cabernet — CHANGES_REQUESTED** [review]

File: .github/workflows/configure-repo.yaml [hunk 1]
```diff
@@ -103,10 +103,14 @@ jobs:
           gh variable set AWS_ACCOUNT_ID --repo ${{ github.repository }} --env=development --body="${{ inputs.aws_account_id_dev }}"
           gh variable set AWS_REGION --repo ${{ github.repository }} --env=development --body="${{ inputs.aws_region_dev }}"
           gh variable set ROLE_TO_ASSUME --repo ${{ github.repository }} --env=development --body="arn:aws:iam::${{ github.event.inputs.aws_account_id_dev }}:role/${{ github.event.inputs.project_name }}-GitHubActionsRole"
+          gh variable set PYTHON_TF_INIT_ROLE_TO_ASSUME --repo ${{ github.repository }} --env=development --body="arn:aws:iam::${{ github.event.inputs.aws_account_id_dev }}:role/Python-TF-Init-GitHubActionsRole"
```
[the remark this answers is not in the document]
Comment (canle144) [reply .github/workflows/configure-repo.yaml]: removed

File: .github/workflows/terraform-python-init.yaml [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+          if aws iam get-role --role-name $github_action_role > /dev/null 2>&1; then
+            echo "[INFO] IAM role $github_action_role already exists."
+          else
+            echo "[INFO] Creating IAM role $github_action_role"
+            aws iam create-role \
+            --role-name $github_action_role \
+            --assume-role-policy-document file://$policy_file 
+          fi
+
+      - name: Create policies & attatch to GitHubActionsRole
```
Comment (brook-cabernet) [inline .github/workflows/terraform-python-init.yaml:148]: nice

File: .github/workflows/terraform-python.yaml [hunk 3]
```diff
@@ -43,11 +43,6 @@ permissions:
 jobs:
   terraform:
     runs-on: ubuntu-latest
-
-    defaults:
-      run:
-        working-directory: ./tf/main
```
Comment (brook-cabernet) [inline .github/workflows/terraform-python.yaml:49]: this will break existing projects, so we need to backfill existing projects with a version tag?

**Review by brook-cabernet — APPROVED** [review]

**Pull request merged**